metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -221,6 +221,20 @@ void add_reverse_complements(size_t k, size_t num_threads, Vector<T> *kmers) {
     ips4o::parallel::sort(kmers->begin(), kmers->end(), utils::LessFirst(), num_threads);
 }
 
+
+template <typename T_TO, typename T_FROM>
+ChunkedWaitQueue<T_TO>& reinterpret_container(ChunkedWaitQueue<T_FROM> &container) {
+    static_assert(sizeof(T_TO) == sizeof(T_FROM));
+    return reinterpret_cast<ChunkedWaitQueue<T_TO>&>(container);
+}
+
+template <typename T_TO, typename T_FROM>
+Vector<T_TO>& reinterpret_container(Vector<T_FROM> &container) {
+    static_assert(sizeof(T_TO) == sizeof(T_FROM));
+    return reinterpret_cast<Vector<T_TO>&>(container);
+}
+
```

Comment:
I know these are not in the public interface, but maybe add a static assert to these to make sure that `T_TO` and `T_FROM` are not `bool` in case this code ends up being used in other places.

Reply:
done